### PR TITLE
Include additional cookie for Jira cloud auth to solve Chrome issue

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

Since browsers are improving their security around third-party cookies, we can no longer assume the user's browser will have access to their MM session token when using the Jira cloud iframe when connecting their Jira account to MM. This PR makes an additional cookie for a one-time use during the authentication process, with the `SameSite` setting set to `off`.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-jira/issues/606
Fixes https://github.com/mattermost/mattermost-plugin-jira/issues/634

### CODE REVIEW

**Comment by mickmister** [comment]
@hanzei I'm not sure what's going on with the `go.mod` dependencies for the linter. Do you have any ideas?

```diff
 	github.com/circonus-labs/circonusllhist v0.1.3
 	github.com/dghubble/oauth1 v0.5.0
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
+	github.com/go-ldap/ldap v3.0.3+incompatible // indirect
+	github.com/go-redis/redis v6.15.9+incompatible // indirect
 	github.com/google/uuid v1.1.1
 	github.com/mattermost/mattermost-plugin-api v0.0.12-0.20200908143138-66edf222f7ea
 	github.com/mattermost/mattermost-plugin-autolink v1.1.3-0.20200203183014-8c82b7dc7fa6
 	github.com/mattermost/mattermost-plugin-workflow-client v0.0.0-20200225165807-b0524a245156
+	github.com/mattermost/mattermost-server v5.11.1+incompatible
 	github.com/mattermost/mattermost-server/v5 v5.24.0
+	github.com/nicksnyder/go-i18n v1.10.1 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/rbriski/atlassian-jwt v0.0.0-20180307182949-7bb4ae273058
 	github.com/rudderlabs/analytics-go v3.2.1+incompatible
```

File: server/instance_cloud.go [hunk 1]
```diff
@@ -20,6 +22,7 @@ import (
 	"github.com/mattermost/mattermost-plugin-jira/server/expvar"
 	"github.com/mattermost/mattermost-plugin-jira/server/utils"
 	"github.com/mattermost/mattermost-plugin-jira/server/utils/types"
+	"github.com/mattermost/mattermost-server/model"
```
Comment (hanzei) [inline server/instance_cloud.go:25]: This should be 
```suggestion
	"github.com/mattermost/mattermost-server/v5/model"
```

**Comment by aaronrothschild** [comment]
@jfrerich I'm concerned this may affect more users than we expect, can we review this and get it in for 3.0.1? cc @mickmister

**Comment by jfrerich** [comment]
@aaronrothschild, sure.  Let me give a review really quick

**Review by jfrerich — APPROVED** [review]
LGTM.  Thanks for the fix @mickmister!

**Review by jupenur — APPROVED** [review]

**Comment by DHaussermann** [comment]
@hanzei  I will test this shortly. Just to confirm, your comment requires no further action and the `Dev Review` label can be removed?

**Comment by hanzei** [comment]
@DHaussermann No actions required from my side. But @levb should take a look before QA review.

**Review by levb — APPROVED** [review]

**Comment by aaronrothschild** [comment]
@mickmister can we include this in 3.0.1 Milestone?

**Comment by DHaussermann** [comment]
@mickmister I have done a round of testing on this. I still seeing this issue on Safari browser for Mac OS. 
On Safari I still see 
```
Mattermost failed to recognize your user account. Please make sure third-party cookies are enabled in your browser settings. You can disable this setting after conntecting your Jira account. Please also make sure you are signed into Mattermost at https://dkh-local.ngrok.io
```
Is Safari unique in some way that your solution will not work there? I don't see much control for managing cookies manually as a work-around...
Screen Shot 2020-11-18 at 1 42 47 PM
 
Tested on:
- Desktop 4.5.2
- Desktop 4.6.1
- Chrome 
- Firefox 
- Safari

**Comment by mickmister** [comment]
@DHaussermann For Safari, the third-party cookie setting is the one labeled `Prevent cross-site tracking`

**Comment by DHaussermann** [comment]
Thanks @mickmister! Yes, unchecking `Prevent cross-site tracking` resolves the issue.

This means that for Safari, the user must change the stock config to map a user, @aaronrothschild is this something we're okay with if we document it?

**Comment by mickmister** [comment]
> This means that for Safari, the user must change the stock config to map a user

@DHaussermann I don't think that is new behavior, since we relied on third-party cookies to begin with. Did Safari used to work without changing this `Prevent cross-site tracking` browser setting?

**Comment by DHaussermann** [comment]
Thanks @mickmister, you're correct. I checked an another machine running the previous version `13.1.2` and this behavior is not new.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
This is working as expected. Tested on:
- Desktop 4.5.2
- Desktop 4.6.1
- Chrome
- Firefox
- Safari (requires change to stock config unrelated to this change)

Regression tested mapping a user to Jira server as well. No issues found.

LGTM!

**Pull request merged**